Julián

JulianLz96 opened the pull request:
Internacionalización

JulianLz96 committed:
Internacionalizacion

JulianLz96 committed:
UpdateIn

JulianLz96 committed:
Try2

JulianLz96 committed:
try4

JulianLz96 committed:
try4.2

JulianLz96 committed:
try final

JulianLz96 committed:
Internacionalizacion

Dos lenguajes agregados

JulianLz96 committed:
Internacionalización

Update:
-Portugués
-Ingles
-Español
-Francés

AlexisGlez commented on src/gui/controllers/LoginController.java:
```diff
@@ -19,7 +19,8 @@ public void initialize(URL arg0, ResourceBundle arg1) {
 
 	@FXML public void login(ActionEvent event) {
-		Utils.createWindow(null, LoginController.this, "../fxml/Home.fxml", "TasksFX", null, "../css/home.css");
+		ResourceBundle bundle = ResourceBundle.getBundle("resources.i18n.login");
```
This should load the Home resources instead of login.
The line 23 is opening  the Home Window.

AlexisGlez commented on src/gui/fxml/Home.fxml:
```diff
@@ -15,13 +15,13 @@
           <AnchorPane minHeight="0.0" minWidth="0.0">
                <children>
-                  <Button mnemonicParsing="false" onAction="#addFolder" text="+" AnchorPane.rightAnchor="20.0" AnchorPane.topAnchor="10.0" />
+                  <Button mnemonicParsing="false" onAction="#addFolder" text="%home_folderAdd" AnchorPane.rightAnchor="20.0" AnchorPane.topAnchor="10.0" />
```
Let's use snake case for every string property, change `home_folderAdd` to `home_folder_add`

AlexisGlez commented on src/gui/fxml/Home.fxml:
```diff
@@ -15,13 +15,13 @@
           <AnchorPane minHeight="0.0" minWidth="0.0" prefHeight="160.0" prefWidth="100.0">
                <children>
-                  <Button layoutX="348.0" layoutY="17.0" mnemonicParsing="false" onAction="#addTask" text="+" AnchorPane.rightAnchor="20.0" AnchorPane.topAnchor="10.0" />
+                  <Button layoutX="348.0" layoutY="17.0" mnemonicParsing="false" onAction="#addTask" text="%home_taskAdd" AnchorPane.rightAnchor="20.0" AnchorPane.topAnchor="10.0" />
```
Same as above

AlexisGlez commented on src/gui/fxml/Task.fxml:
```diff
@@ -95,7 +95,7 @@
                            </GridPane.margin>
                         </TextArea>
-                        <Label text="Date:" GridPane.halignment="CENTER" GridPane.rowIndex="1" GridPane.valignment="TOP">
+                        <Label text="%task_reminderDate" GridPane.halignment="CENTER" GridPane.rowIndex="1" GridPane.valignment="TOP">
```
Same as above.

AlexisGlez commented on src/gui/fxml/Task.fxml:
```diff
@@ -105,22 +105,22 @@
                            </GridPane.margin>
                         </TextField>
-                        <Label text="Note:" GridPane.halignment="CENTER" GridPane.rowIndex="2" GridPane.valignment="CENTER">
+                        <Label text="%task_reminderNote" GridPane.halignment="CENTER" GridPane.rowIndex="2" GridPane.valignment="CENTER">
```
Same as above.

AlexisGlez commented on src/gui/fxml/Task.fxml:
```diff
@@ -105,22 +105,22 @@
                            </GridPane.margin>
                         </Button>
-                        <Label text="Title:" GridPane.halignment="CENTER" GridPane.valignment="TOP">
+                        <Label text="%task_reminderTitle" GridPane.halignment="CENTER" GridPane.valignment="TOP">
```
Same as above.

AlexisGlez commented on src/resources/i18n/folder.properties:
```diff
@@ -0,0 +1,5 @@
+folder_favorite = Favorite
+folder_accept = Accept
+folder_cancel = Cancel
+folder_Add = Add
```
Change the properties to just use snake case.

AlexisGlez commented on src/resources/i18n/task_fr.propeprties:
```diff
@@ -0,0 +1,12 @@
+task_text = Texte
```
The name of this file has a typo.
Change `src/resources/i18n/task_fr.propeprties` to `src/resources/i18n/task_fr.properties`

AlexisGlez commented on src/utils/Utils.java:
```diff
@@ -13,9 +16,12 @@
-        	FXMLLoader fxmlLoader = new FXMLLoader(parent.getClass().getResource(fxmlLocation));
+        	//Locale locale = new Locale("es", "MX");
+        	//String resourceLocation = "resources.i18n.UIResources";
+        	//ResourceBundle bundle = ResourceBundle.getBundle(resourceLocation, locale);
```
Delete all the useless commented code.

AlexisGlez commented on src/utils/Utils.java:
```diff
@@ -13,9 +16,12 @@
 public class Utils {
 
-	public static void createWindow(Stage stage, Object parent, String fxmlLocation, String sceneTitle, Object userData, String cssLocation, Double sceneWidth, Double sceneHeight) {
+	public static void createWindow(Stage stage, Object parent, String fxmlLocation, String sceneTitle, Object userData, String cssLocation, Double sceneWidth, Double sceneHeight, ResourceBundle bundle) {
```
Instead of receive the Resource Bundle as a parameter, let's receive a String with that contains the bundle to get. Move all the logic of getting the bundle's to this method.

AlexisGlez replied:
Move the bundle String resource before the height and widths params like this:
```public static void createWindow(Stage stage, Object parent, String fxmlLocation, String sceneTitle, Object userData, String cssLocation, String bundleLocation, Double sceneWidth, Double sceneHeight) {```

AlexisGlez commented on src/utils/Utils.java:
```diff
@@ -50,11 +56,11 @@ public void handle(WindowEvent event) {
 
-	public static void createWindow(Stage stage, Object parent, String fxmlLocation, String sceneTitle, Object userData, String cssLocation) {
-		createWindow(stage, parent, fxmlLocation, sceneTitle, userData, cssLocation, null, null);
+	public static void createWindow(Stage stage, Object parent, String fxmlLocation, String sceneTitle, Object userData, String cssLocation, ResourceBundle bundle) {
```
Instead of modifying this method, lets create another one that receives the bundle String.

JulianLz96 committed:
Internacionalización

Update string property.

JulianLz96 committed:
Internacionalización

Fix typo in task_fr.propeprties to task_fr.properties

JulianLz96 committed:
Internacionalización

Update: createWindow con String
Se añadió el texto del botón de Editar en el bundle

AlexisGlez commented on src/gui/controllers/LoginController.java:
```diff
@@ -19,7 +19,7 @@ public void initialize(URL arg0, ResourceBundle arg1) {
 
 	@FXML public void login(ActionEvent event) {
-		Utils.createWindow(null, LoginController.this, "../fxml/Home.fxml", "TasksFX", null, "../css/home.css");
+		Utils.createWindow(null, LoginController.this, "../fxml/Home.fxml", "TasksFX", null, "../css/home.css", "resources.i18n.login");
```
This should be `resources.i18n.home` because we are creating the Home Controller here.

AlexisGlez commented on src/utils/Utils.java:
```diff
@@ -13,9 +16,10 @@
         try {
-        	FXMLLoader fxmlLoader = new FXMLLoader(parent.getClass().getResource(fxmlLocation));
+        	ResourceBundle bundle = ResourceBundle.getBundle(bundleLocation);
+          	FXMLLoader fxmlLoader = new FXMLLoader(parent.getClass().getResource(fxmlLocation), bundle);
```
If we send `bundleLocation` as null this may crash, lets change line 21 and 22 like this:
```
URL fxmlUrl = parent.getClass().getResource(fxmlLocation);
FXMLLoader fxmlLoader = bundleLocation != null ? bundleLocation new FXMLLoader(fxmlUrl, ResourceBundle.getBundle(bundleLocation)) : new FXMLLoader(fxmlUrl);
```

AlexisGlez reviewed (changes requested):
Nice job!
Just 2 more comments 👍

JulianLz96 committed:
Remove Login

Looseness added

AlexisGlez commented on src/gui/Main.java:
```diff
@@ -11,19 +13,8 @@
-
-				}
-			}
+		Utils.createWindow(primaryStage, Main.this, "./fxml/Home.fxml", "TasksFX", null, "./css/Home.css", "resources.i18n.Home");
```
The Home CSS and the resource should be lower case:
`./css/Home.css` -> `./css/home.css`
`resources.i18n.Home` -> `resources.i18n.home`

AlexisGlez commented:
I'm running the application, when I click `Edit` one task it crashes:
![image](https://user-images.githubusercontent.com/13372889/31557660-5055a23c-b00f-11e7-9d49-5f1eced2a7fb.png)

@JulianLz96 ^^^

JulianLz96 committed:
Internacionalización

Lower case resources

JulianLz96 committed:
Merge branch 'master' into Julián

JulianLz96 committed:
Internacionalización

Fix error

AlexisGlez committed:
Add missing properties and fix typo
